### PR TITLE
Merge 3.x into Master

### PR DESCRIPTION
**Pull request by VincentLanglet**
.

### CODE REVIEW

File: tests/Action/AppendFormFieldElementActionTest.php [hunk 1]
```diff
@@ -90,25 +82,26 @@ public function testAppendFormFieldElementAction(): void
             'context' => 'list',
         ], [], [], [], [], ['REQUEST_METHOD' => Request::METHOD_POST]);
 
-        $modelManager = $this->prophesize(ModelManagerInterface::class);
+        $modelManager = $this->createStub(ModelManagerInterface::class);
         $formView = new FormView();
-        $form = $this->prophesize(Form::class);
+
```
Comment (jordisala1991) [inline tests/Action/AppendFormFieldElementActionTest.php:87]: this space looks strange

**Comment by VincentLanglet** [comment]
I just need to fix the BreadcrumbsBuilderTest

**Review by jordisala1991 — COMMENTED** [review]
The breadcrumb builder changed a lot, it is failing at some point because using prophecy

**Comment by VincentLanglet** [comment]
> The breadcrumb builder changed a lot, it is failing at some point because using prophecy

Yes, I push with the conflict if someone wants to fix it.
There was a new test https://github.com/sonata-project/SonataAdminBundle/blob/master/tests/Admin/BreadcrumbsBuilderTest.php#L33 in master

File: src/Resources/views/CRUD/base_list.html.twig [hunk 2]
```diff
@@ -196,6 +196,12 @@ file that was distributed with this source code.
                                                         <i class="fa fa-arrow-circle-o-down" aria-hidden="true"></i>
                                                         {{ ("export_format_" ~ format)|trans({}, 'SonataAdminBundle') }}
                                                     </a>
+                                                    {#
+                                                    <a href="{{ admin.generateUrl('export', admin.datagrid.paginationparameters(0) + {'format' : format}) }}">
+                                                        <i class="fa fa-arrow-circle-o-down" aria-hidden="true"></i>
+                                                        {{ ("export_format_" ~ format)|trans({}, 'SonataAdminBundle') }}
+                                                    </a>
+                                                    #}
```
Comment (VincentLanglet) [inline src/Resources/views/CRUD/base_list.html.twig:204]: This is weird

Comment (jordisala1991) [reply src/Resources/views/CRUD/base_list.html.twig]: why? this was the code to uncomment on next major?

Comment (VincentLanglet) [reply src/Resources/views/CRUD/base_list.html.twig]: It was already fixed on master, this comment is the same that the line before.

**Comment by jordisala1991** [comment]
Maybe we should start with the original version of this test on master and improve on next PR. I reviewed the rest of this merge and looks okay, needs fix php cs tho.

**Comment by VincentLanglet** [comment]
> Maybe we should start with the original version of this test on master and improve on next PR. I reviewed the rest of this merge and looks okay, needs fix php cs tho.

I took the version of the test in 3.x and fix phpunit warnings, it should be ok.

**Comment by VincentLanglet** [comment]
It's hard to focus on Sonata 4.0 when it takes 2h to merge 3.x into master 😕

**Comment by VincentLanglet** [comment]
Green tests @jordisala1991 :)

**Comment by jordisala1991** [comment]
> It's hard to focus on Sonata 4.0 when it takes 2h to merge 3.x into master 😕

Completely agree.. that's why I wanted to do some sort of feature freeze on 3.x. 🤷‍♂️

**Review by jordisala1991 — APPROVED** [review]
Looks good, thank you for taking care of the merge

**Pull request merged**